### ISSUE TITLE
Bad PID ''; expected '112' 

### ISSUE THREAD

**Issue by tctfox** [jkread/arkserver#3]
Hello there

When I try to build the docker for the Ark Server I get an error. I get this error when I use the docker compose file and when I use the docker run command. 

I never got this error with the docker image from  thmhoag  

https://pastebin.com/4JAjNiUc

Here is the error. I used your docker compose file and this command here:

docker run -d -v steam:/home/steam -v ark:/ark -p 27015:27015 -p 27015:27015/udp -p 7778:7778 -p 7778:7778/udp -p 7777:7777 -p 7777:7777/udp jkread/arkserver

**Comment by YodaDaCoda** [comment]
I've been getting basically the same error (different PIDs as I leave it running). I think this is because there's not enough memory on my computer (4GB available to the VM I'm running this in), and the server is not able to load all the resources and dies from OOM. Unfortunately I'm not able to confirm as I'm not sure how to get logs.

**Comment by tctfox** [comment]
The thing is, the server crashes and never starts up. It doesnt happen with the other docker image so I dont think its a ram issue. Also I have 64gb ram on my server

**Comment by motey** [comment]
see https://github.com/jkread/arkserver/issues/5

**Comment by jkread** [comment]
Are the ark and steam volumes pre-existing or created as part of you first run? I'm wondering if you are missing all the files.

**Comment by YodaDaCoda** [comment]
For me, created as part of first run. There was nothing in the ark/steam volumes prior to my first run and there's stuff there now.

**Comment by tctfox** [comment]
It should definitely create the volumes and files by itself. But for me no matter how many times I run it I get errors

**Comment by jkread** [comment]
So during the first run it is supposed to download all the files needed. While it says it is it isn't actually downloading the files.  These two lines mean necessary files are missing. Try removing the volumes and starting the container and see if it downloads everything it should. Also check the permissions on the volumes if that doesn't work. I just put out a new image that fixes a few issues.

/usr/local/bin/arkmanager: line 671: /home/steam/.steam/steamcmd/steamcmd.sh: No such file or directory 
/usr/local/bin/arkmanager: line 1267: /ark/server/ShooterGame/Binaries/Linux/ShooterGameServer: No such file or directory

**Comment by YodaDaCoda** [comment]
I've cleaned out the volumes and re-created them. I've also done a `docker pull jkread/arkserver` and it has pulled updates, I'm up-to-date at time of writing.

```
Using default tag: latest
latest: Pulling from jkread/arkserver
Digest: sha256:2e37803e36525714c76d7b05cdf36c94bf27095629a0645e990842f590d2c17b
Status: Image is up to date for jkread/arkserver:latest
docker.io/jkread/arkserver:latest
```

Log of the docker container starting up:
https://pastebin.ubuntu.com/p/xHgsW9xFr5/

**Comment by jkread** [comment]
I'm starting to think the folder permissions are messed up somehow. Can you run tree on the ark and steam volume locations from your host machine?

**Comment by YodaDaCoda** [comment]
This is what I get after I `rm -rf`'d the ark directory and re-created the container. All the files/folders are owned by 1000:1000 (which is the `will` user/group you see in the output). My steam volume is mounted inside my ark volume just for ease of management.

```
$ tree -apugif ark
ark
[drwxr-xr-x will     will    ]  ark/backup
[drwxr-xr-x will     will    ]  ark/config
[-rw-r--r-- will     will    ]  ark/config/arkmanager.cfg
[-rw-r--r-- will     will    ]  ark/config/crontab
[drwxr-xr-x will     will    ]  ark/log
[-rw-r--r-- will     will    ]  ark/log/arkmanager.log
[drwxr-xr-x will     will    ]  ark/server
[drwxr-xr-x will     will    ]  ark/server/ShooterGame
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Binaries
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Binaries/Linux
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Content
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Content/Mods
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Saved
[-rw-r--r-- will     will    ]  ark/server/ShooterGame/Saved/.arkmanager-main.pid
[-rw-r--r-- will     will    ]  ark/server/ShooterGame/Saved/.arkserver-main.pid
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Saved/Config
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Saved/Config/LinuxServer
[drwxr-xr-x will     will    ]  ark/server/ShooterGame/Saved/SavedArks
[-rw-r--r-- will     will    ]  ark/server/.startAfterUpdate-main
[drwxr-xr-x will     will    ]  ark/staging
[drwxr-xr-x will     will    ]  ark/steam
[-rw-r--r-- will     will    ]  ark/steam/.bash_logout
[-rw-r--r-- will     will    ]  ark/steam/.bashrc
[-rw-r--r-- will     will    ]  ark/steam/.profile
[-rw-r--r-- will     will    ]  ark/steam/.sudo_as_admin_successful

15 directories, 10 files
```

**Comment by jkread** [comment]
Can you post your docker run or compose?  There should be a ton more files.

Actually that might be part of the issue. Mount the steam directory as its own volume, I don't believe the nesting will work.

I'm not sure how you even managed to nest volumes, maybe you are using bind points.

**Comment by jkread** [comment]
Actually that might be part of the issue. Mount the steam directory as its own volume, I don't believe the nesting will work.

**Comment by YodaDaCoda** [comment]
Yep I'm using bind mounted volumes. I've used a similar strategy before on other containers without issue. Tried making the steam mount its own separate thing without any change in output.

I'm using compose in json format (my brain struggles with yaml).

compose file:
https://pastebin.ubuntu.com/p/J9P9R9KDm6/

container log:
https://pastebin.ubuntu.com/p/pQKzKW67rv/

**Comment by jkread** [comment]
Ok using your compose I was able to recreate the issue. Working on it now.

**Comment by jkread** [comment]
I believe this to be resolved in the current latest build. Please test and let me know. May require a clearing of volumes or binds.

**Comment by YodaDaCoda** [comment]
Wonderful, the server appears to start correctly now (as best as I can tell)! It's not showing up in the online server list, but I think that's due to configuration on my side.

**Comment by jkread** [comment]
Yeah I had some tweaking to do the first time I added a dedicated server to Ark. Don't really remember what it finally was. I'll try to look and see.
